src/extremeweatherbench/data_loader.py

```diff
@@ -0,0 +1,169 @@
+import xarray as xr
+from extremeweatherbench import config, utils
+import pandas as pd
+import logging
+import dataclasses
+from typing import Tuple, Optional
```

Comment:
nit: import order; be sure that `isort` is installed and enabled in your dev environment. It will group imports by groups and then sort alphabetically. Makes it easier to find things.